src/sas/system/user.py

```diff
@@ -124,3 +137,140 @@ def copy_old_files_to_new_location():
             shutil.copy2(old_path, new_path)
 
+
+def module_copytree(module: str, src: PATH_LIKE, dest: PATH_LIKE) -> None:
+    """Copy the tree from a module to the specified directory
+
+    module: name of the Python module (the "anchor" for importlib.resources)
+    src: source name of the resource inside the module
+    dest: destination directory for the resources to be copied into; will be
+        created if it doesn't exist
+    """
+    spth = Path(src)
+    src = str(spth)
+
+    dpth = Path(dest)
+    dpth.mkdir(exist_ok=True, parents=True)
+
+    for resource in importlib.resources.files(module).joinpath(src).iterdir():
+        f_name = dpth / resource.name
+        if "__pycache__" in resource.name:
+            continue
+
+        if resource.is_dir():
+            # recurse into the directory
+            module_copytree(module, spth / resource.name, dpth / resource.name)
+        elif resource.is_file() and not f_name.exists():
+            logging.debug("Copied: %s", spth / resource.name)
+            with open(dpth / resource.name, "wb") as dh:
+                dh.write(resource.read_bytes())
+        else:
+            logging.warning("Skipping %s (unknown type)", spth / resource.name)
+
+
+def copy_module_resources() -> bool:
+    """Obtain the source and build output from within the installed sas module"""
+
+    # Look in the module for the resources. We know that there is a conf.py
+    # for sphinx so check that it exists; checking that the file exists and
+    # not just the directory protects against empty directories
+    if importlib.resources.files("sas").joinpath("docs-source/conf.py").is_file():
+        logging.info("Extracting docs from sas module")
+        module_copytree("sas", "docs-source", MAIN_DOC_SRC)
+        module_copytree("sas", "docs", MAIN_BUILD_SRC / "html")
+        module_copytree("sas", "example_data", EXAMPLE_DATA_DIR)
+        return True
+
+    return False
+
+
+def locate_unpacked_resources() -> tuple[Path, Path]:
+    """Locate the resources unpacked on disk"""
+    # Look near where sasview executable sits - if it's from the pyinstaller
+    # bundle or from run.py then the doc source will be close by. Note that
+    # this won't be true for POSIX-like installations where the executable
+    # is in /usr/bin, ~/.local/bin, or .../venv/bin.
+    exe_dir = Path(sys.argv[0]).parent
+
+    if (exe_dir / "doc").exists():
+        # This is the directory structure for the installed version of SasView
+        # such as when installed from the pyinstaller bundle prior to v6.1
+        source_dir = exe_dir / "doc" / "source"
+        build_dir = exe_dir / "doc" / "build"
+
+    elif (exe_dir.parent / "Frameworks" / "doc").exists():
+        # In the MacOS bundle, the executable and packages are in parallel directories
+        source_dir = exe_dir.parent / "Frameworks" / "doc" / "source"
+        build_dir = exe_dir.parent / "Frameworks" / "doc" / "build"
+
+    else:
```

Comment:
It would probably be safer to `elif` this branch and use `else` to throw or at least send some info to warn the user that `source_dir` and `build_dit` can't be found.